inst/sql/sql_server/CreateConceptCountTable.sql

```diff
@@ -16,82 +17,152 @@ INTO @concept_counts_table
 } : { 
 INTO @work_database_schema.@concept_counts_table
 }
+
+{@use_achilles} ? {
+
+-- Achilles analysis ids used for creating the concept counts table
+-- condition:    400 (persons),  401 (standard concepts),  425 (source concepts)
+-- drug:         700 (persons),  701 (standard concepts),  725 (source concepts)
+-- procedure:    600 (persons),  601 (standard concepts),  625 (source concepts)
+-- measurement: 1800 (persons), 1801 (standard concepts), 1825 (source concepts)
+-- observation:  800 (persons),  801 (standard concepts),  825 (source concepts)
+FROM (
+	SELECT 
+		CAST(stratum_1 AS INT) AS concept_id,
+		count_value AS concept_count
+	FROM @results_database_schema.achilles_results
+	WHERE analysis_id IN (401,601,701,801,1801,425,625,725,825,1825) AND stratum_1 != '0'
+) q1
+LEFT JOIN 
+(
+	SELECT 
+		CAST(stratum_1 AS INT) AS concept_id,
+		count_value AS concept_subjects
+	FROM  @results_database_schema.achilles_results
+	WHERE analysis_id IN (400,600,700,800,1800) AND stratum_1 != '0'
+) q2
+ON q1.concept_id = q2.concept_id
+
+} : {
+
 FROM (
 	SELECT condition_concept_id AS concept_id,
 		COUNT_BIG(*) AS concept_count,
 		COUNT_BIG(DISTINCT person_id) AS concept_subjects
-	FROM @cdm_database_schema.condition_occurrence
+	FROM @cdm_database_schema.condition_occurrence co
+	JOIN @cdm_database_schema.observation_period op 
```

Comment:
I don't have a particularly strong opinion on forcing the concepts to occur inside the observation windows (it makes perfect sense for this to be a rule) but this will change the results of the package. However, maybe this should be parameterized to allow users to create consistent reports?

I'd like to hear some thoughts here before making a decisions one way or the other

Reply:
My impression is that when people create cohorts with Atlas they are most often using concepts inside the observation period. In the case where a CDM has many concepts outside observation periods it might be confusing to display concepts outside observation period in CohortDiagnostics which are not picked up by Atlas cohort definitions by default. It should be clear that if someone is expecting Atlas to pick up the same number of records they see in CohortDiagnostics they will need include concepts outside observation period. The whole notion of concepts outside observation period is confusing even though I do understand there are legitimate arguments for it. It is hard to explain to OHDSI newcomers: "Observation time is when we see events but there are also events that occur outside observation time that you may want to include". It's strange because if you strictly interpret the definition of observation time there should not be observations outside of it.